ui/components/src/Table/TableCheckbox.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Checkbox, CheckboxProps, alpha } from '@mui/material';
+import { TableDensity } from './model/table-model';
+
+export interface TableCheckboxProps extends Pick<CheckboxProps, 'checked' | 'indeterminate' | 'onChange'> {
+  color?: string;
+  density: TableDensity;
+}
+
+export function TableCheckbox({ color, density, ...otherProps }: TableCheckboxProps) {
+  const isCompact = density === 'compact';
+
+  return (
+    <Checkbox
+      size={isCompact ? 'small' : 'medium'}
+      {...otherProps}
+      // Disable ripple and set background color below because of some issues
+      // with re-rendering with the keyboard interactions that causes the ripple
+      // animation to stutter.
```

Comment:
I spent a bunch of time debugging this and could not figure it out. Keyboard users don't care about the ripple and it works well with this workaround, so this is another "let's revisit this later because it's working well enough for now" case.